keepalived/check/check_daemon.c

```diff
@@ -217,6 +219,7 @@ reload_check_thread(__attribute__((unused)) thread_t * thread)
 #ifdef _WITH_VRRP_
 	kernel_netlink_close();
 #endif
+	script_killall(master, SIGHUP);
```

Comment:
This sends a SIGTERM to the child process if keepalived is stopping, and a SIGHUP to the child process if the config is being reloaded. Whichever of these two situations is occurring, does the child process need to know the difference. All we want is that it stops, although it might be possible that a child process wants to do something different depending on whether keepalived is stopping or reloading (but I can't think of a reason why), but then the problem is that if the child receives a SIGTERM, it can't distinguish between the keepalived process terminating and the child has been running for too long.

The same applies further down in `misc_check_child_thread()` and `stop_vrrp()`.